tests/test_connection.py

```diff
@@ -406,7 +412,10 @@ def test_uri_without_credentials_doesnt_override_conn_settings(self):
         # w/ the provided username/password and failed - that's the desired
         # behavior. If the MongoDB URI would override the credentials
         with pytest.raises(OperationFailure):
-            get_db()
+            db = get_db()
+            # pymongo 4.x does not call db.authenticate and needs to perform an operation to trigger the failure
+            if PYMONGO_VERSION >= (4,):
```

Comment:
minor remark here but it's better to test the 2 things separately (otherwise we can't ensure that it makes it this far)

```
        if PYMONGO_VERSION >= (4,):
            db = get_db()
            with pytest.raises(OperationFailure):
                # pymongo 4.x does not call db.authenticate and needs to perform an operation to trigger the failure
                db.list_collection_names()
        else:
            with pytest.raises(OperationFailure):
                get_db()
```

Reply:
done